fvm/fvm.go

```diff
@@ -140,14 +140,27 @@ func (vm *VirtualMachine) RunV2(
 		WithComputationLimit(uint(proc.ComputationLimit(ctx))).
 		WithMemoryLimit(proc.MemoryLimit(ctx))
 
-	meterParams, err = getEnvironmentMeterParameters(
-		ctx,
-		v,
-		txnPrograms,
-		meterParams,
-	)
-	if err != nil {
-		return fmt.Errorf("error gettng environment meter parameters: %w", err)
+	if ctx.meterParameters != nil {
```

Comment:
correct, the `if ctx.meterParameters != nil {` should be only called by `computer` which is Flow itself, as long all EN nodes are deployed simultaneously, there should be no spork disagreement btw nodes, and ideally VN nodes should be deployed at the same time as EN too, to avoid verification failures. So wondering if the deployment of this breaking change will require some manual handling during the next spork.

and for the `if ctx.meterParameters == nil {` branch specifically for emulator, maybe we can leave it here for now, so that before emulator change is deployed, emulator can still function as usual. also with or without this branch, for non-emulator users the spork will be different anyway.

Reply:
Regardless of how/when the meter parameters is fetched, i would much rather that don't skip the state replay.  The current approach (skipping replay) works fine only when transactions are executed sequentially.  Skipping the replay (i.e., not making the transaction self contained) makes it impossible to check for read-write conflict in parallel execution (since you're omitting the reads).

Btw, there's no significant advantage in doing at a collection level since collections tend to have very few transactions today.  We're better off doing it at the block level.

Reply:
to be clear, i actually don't care that much about the spock hash.  what I care about is the accurate bookkeeping of the transaction's read set (Having the correct spock is just a side benefit for maintaining accurate bookkeeping).